docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,423 @@
+# 2-Node Active-Active HA using Dqlite
+
+## Rationale
+
+High availability is a mandatory requirement for most production-grade
+Kubernetes deployments, usually implying three or more nodes.
+
+However, 2-node HA clusters are desired in some situations due to cost saving
+and operational efficiency considerations. Follow this guide to learn how
+Canonical Kubernetes can achieve high availability with just two nodes
+while using the default datastore, Dqlite.
+
+Dqlite cannot achieve Raft quorum with less than three nodes. This means that
+Dqlite will not be able to replicate data and the secondaries will simply
+forward the queries to the primary node.
+
+In the event of a node failure, the database will have to be recovered by
+following the steps outlined in the [Dqlite recovery guide].
+
+## Proposed solution
+
+Since Dqlite data replication is not available in this situation, we propose
+using synchronous block level replication through DRBD.
+
+The cluster monitoring and failover process will be handled by Pacemaker and
+Corosync. In the event of a node failure, the DRBD volume will be mounted on
+the replica, which can then access the most recent version of the Dqlite database.
+
+Additional recovery steps are automated and invoked through Pacemaker.
+
+## Alternatives
+
+Another possible approach is to use PostgreSQL with Kine and logical replication.
+However, it is outside the scope of this document.
+
+See the [external datastore guide] for more information on how Canonical
+Kubernetes can be configured to use other datastores.
+
+## Guide
+
+### Prerequisites
+
+Make sure that:
+
+* Both nodes have joined the Kubernetes cluster.
+  See the [getting started] and [add/remove nodes] guides.
+* The user associated with the HA service has SSH access to the peer node and
+  passwordless sudo configured. For simplicity, the default "ubuntu" user can
+  be used.
+* We recommend using static IP configuration.
+
+The [2ha.sh script] automates most operations related to the 2-node HA scenario.
+Retrieve it like so:
+
+```
+sudo mkdir -p /var/snap/k8s/common
+repo=https://raw.githubusercontent.com/petrutlucian94/k8s-snap
```

Comment:
This script should be part of the snap. If you put it into hack then it should either be available at `/snap/k8s/current/k8s/hack` or you would need to update the `snapcraft.yaml` to add this directory to the snap.

Reply:
Great, I guess we can just use ``/snap/k8s/current/k8s/hack/2ha.sh``.